stellargraph/mapper/sequences.py

```diff
@@ -527,3 +528,45 @@ def __len__(self):
 
     def __getitem__(self, index):
         return self.inputs, self.targets
+
+
+class CorruptedNodeSequence(Sequence):
+    """
+    Keras compatible data generator that wraps a FullBatchSequence, SparseFullBatchSequence, or
+    RelationalFullBatchNodeSequence and provides corrupted data for training Deep Graph Infomax.
+
+    Args:
+        base_generator: the uncorrupted Sequence object.
+    """
+
+    def __init__(self, base_generator):
+
+        if not isinstance(
+            base_generator,
+            (
+                FullBatchSequence,
+                SparseFullBatchSequence,
+                RelationalFullBatchNodeSequence,
```

Comment:
Have you tested this implementation with RGCN? Because if not, then shall we wait to add this when we are ready to extend to HINs?

Reply:
yeah I think that makes sense to land all the heterogenous infomax stuff together